File: Models/Group.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackupSystem.Models;

public partial class Group
{
    public int GroupId { get; set; }

    public string? GroupName { get; set; }

    public virtual ICollection<StationConfiguration> StationConfigurations { get; set; } = new List<StationConfiguration>();

    public virtual ICollection<StationGroup> StationGroups { get; set; } = new List<StationGroup>();

    [ForeignKey("GroupId")]
    public virtual ICollection<Station> Stations { get; set; } = new List<Station>();

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

}

File: Frontend/src/app/models/group.ts
import { Station } from './station';
import { Config } from './config';

export class Group {
  public groupId: number;

  public groupName: string;

  public stations: Station[];

  public configs: Config[];

  public constructor(
    id: number = 0,
    name: string = '',
    stations: Station[] = [],
    configs: Config[] = []
  ) {
    this.groupId = id;
    this.groupName = name;
    this.stations = stations;
    this.configs = configs;
  }
}
